docs/_sources/week02/display-reports/index.rst.txt

```diff
@@ -0,0 +1,158 @@
+:orphan:
+
+.. _week2_display-reports:
+
+==========================================
+Project Week2: Display Zika Reports on Map
+==========================================
+
+We have an empty interactive map from OSM.
+
+We need to create a new Layer in OpenLayers that will represent our Zika report data.
+
+This will require a fair amount of work, luckily it is primarly things we did in our `Airwaze project <../../studios/airwaze/>`_.
+
+Tasks:
+    - Design Report.java class
+    - Create ReportTests.java
+    - Report table is populated from CSVs
+    - Create ReportTests.java
+    - Create ReportController.java
+    - Create ReportControllerTests.java
+    - Create ReportRepository.java
+    - Create ReportRepositoryTests.java
+    - Amend script.js
+    - Verify Zika report layer renders
+    - Commit
+
+Design Report.java Class
+------------------------
+
+Before we display our report data as a layer on our map in OSM we will first have to transform our data into a state we can work with.
+
+Ultimately OpenLayers is expecting a source of data in a GeoJSON format. Right now our data is locked away in various CSV files.
+
+Let's consider our data's journey.
+
+It currently exists in CSV format, we will have to COPY our CSVs into a PostGIS table as records, our web application will convert records from PostGIS into a Java Object, our web app will convert Java Objects into GeoJSON. The GeoJSON is what OpenLayers is expecting to make a new layer from.
+
+In order to COPY our CSVs into a PostGIS table, the table must exist, and it's columns must match the SQL COPY statement. We have been using Hibernate to create our PostGIS tables for us, Spring scans our project for the @Entity annotation, and automatically creates a table based on the properties of that class. In order to create our table we will first need to design our Report class!
+
+Look over the `provided CSV files <https://gitlab.com/LaunchCodeTraining/zika-data>`_, look for their **shared** columns. These shared columns will be the columns in our table, and we need to create properties on our Report class to match these columns.
+
+Look at your solution to the `Airwaze Studio <../../studios/airwaze/>`_. Take note of how the ``Airports.csv`` file matches up with the ``models/Airport.java`` file, and how the ``routes.csv`` file matches up with the ``models/Route.java`` file.
+
+.. hint::
+
+   Hibernate uses the snake case naming strategy. You can explicitlly define a Column name by using the Spring Data @Column annotation. Above the property declaration you can include the @Column(name = "columnName") annotation and Hibernate will name the column based on the value mapped to the name variable.
+
+After you create your Report.java class re-run ``bootRun``, and then check your database. Your report table still won't have any records in it, but 
+
+Create ReportTests.java
+-----------------------
+
+When creating a new class in Java we should test this file.
```

Comment:
How do you feel about telling the specifically which test function to create? I looked at your solution and testing the constructor is a good idea.

Reply:
I like telling them to create a testReportConstructor method, and to create a new testReport object, but I don't want to provide them with the code for doing that. I want them to look back at the Unit Testing materials to figure out how to create it themselves. I also added a note about not testing code written by third parties, like constructors, but that they were expected to in this class.